flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/ExtractFunctionITCase.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.functions;
+
+import org.apache.flink.table.functions.BuiltInFunctionDefinitions;
+
+import java.time.LocalDateTime;
+import java.util.stream.Stream;
+
+import static org.apache.flink.table.api.DataTypes.BIGINT;
+import static org.apache.flink.table.api.DataTypes.TIMESTAMP;
+
+/** Test {@link BuiltInFunctionDefinitions#EXTRACT} and its return type. */
+class ExtractFunctionITCase extends BuiltInFunctionTestBase {
+    @Override
+    Stream<TestSetSpec> getTestSetSpecs() {
+        return Stream.of(
+                TestSetSpec.forFunction(BuiltInFunctionDefinitions.EXTRACT)
+                        .onFieldsWithData(
+                                LocalDateTime.of(2000, 1, 31, 11, 22, 33, 123456789),
+                                LocalDateTime.of(2020, 2, 29, 1, 56, 59, 987654321),
+                                null)
+                        .andDataTypes(
+                                TIMESTAMP().nullable(),
+                                TIMESTAMP().nullable(),
+                                TIMESTAMP().nullable())
+                        .testSqlResult(
```

Comment:
Why are you testing only sql? Shouldn't this be supported by table api as well?

Reply:
It looks not for every timeunit table api is supported
I will have a look if it is possible to support others

Reply:
> I will have a look if it is possible to support others

Thank you, We should align the two as much as we can, and in this particular case it seems as easy as just adding the missing enum variants

Reply:
I added support for all except `DOW`, `ISODOW`, `DOY`, `ISOYEAR`.
For those 4 I have a question: from one side `extract` is supported for `TimeIntervalUnit` i.e. need to add these 4 elements there, from the other side these 4 elements are not interval units... Does it make sense add extract for `TimePointUnit` e.g. or leave it like it is?

Reply:
What Calcite is doing for those?